src/mmap.rs

```diff
@@ -116,14 +118,17 @@ pub fn check_file_offset(
     file_offset: &FileOffset,
     size: usize,
 ) -> result::Result<(), MmapRegionError> {
-    let file = file_offset.file();
+    let mut file = file_offset.file();
     let start = file_offset.start();
 
     if let Some(end) = start.checked_add(size as u64) {
-        if let Ok(metadata) = file.metadata() {
-            if metadata.len() < end {
-                return Err(MmapRegionError::MappingPastEof);
-            }
+        let filesize = file
+            .seek(SeekFrom::End(0))
+            .map_err(MmapRegionError::SeekEnd)?;
+        file.seek(SeekFrom::Start(0))
```

Comment:
Hmm, do we actually need to reset the position by going back to the start?

Reply:
It's not strictly needed, since the next thing we'll do on the file is create the mapping. But I thought it was more `future-proof` if we returned the fd it to the expected state. I'm fine dropping it either.